### PR TITLE
MAINT: Update parasail-python to 1.1.16

### PR DESCRIPTION
**Pull request by jvkersch**
@bioconda/core Could I ask for a review? This is not a new recipe, but it is my first PR to this repo and I wanted to make sure everything is OK. Thanks!

----------------

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [ ] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

File: recipes/parasail-python/meta.yaml [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
+{% set version = "1.1.16" %}
 
 package:
   name: "{{ name|lower }}"
   version: "{{ version }}"
 
 source:
   url: https://github.com/jeffdaily/parasail-python/archive/v{{ version }}.tar.gz
-  md5: 109fc497924030eebda60bde33f0976b
+  md5: aacd7de464f24ec64daaae2c12ca8c6c
```
Comment (druvus) [inline recipes/parasail-python/meta.yaml]: Could you swap to sha256?

File: recipes/parasail-python/meta.yaml [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
 
 build:
-  number: 2
+  number: 1
```
Comment (druvus) [inline recipes/parasail-python/meta.yaml]: Should start at 0 for new version

**Comment by jvkersch** [comment]
@druvus Thank you for the swift reply! I made the changes you suggested, but there is a build failure due to a linker incompatibility. I think this is due to Parasail prepending `/usr/bin` to the PATH in `setup.py`. I will need some time to properly test for this, and to make a patch.

**Comment by jvkersch** [comment]
I got the Linux build to work (the recipe was lacking a dependency on the C++ compiler, which caused the linking issues) but there's something wrong with the Mac OS build. On Mac OS, parasail seems to insist on downloading and installing the libtool toolchain, I am not sure that that is necessary in a controlled environment like the Bioconda build environment.

**Comment by jvkersch** [comment]
In the end I did have to patch `setup.py` on Mac OS to account for a missing tool name (libtool). My assumption is that the parasail-python author was relying on the Homebrew-installed version of these tools, all of which start with a 'g' prefix to distinguish them from the system versions.

@druvus Would you have time for another round of reviews?

**Comment by jvkersch** [comment]
@druvus Would you mind giving this another look? Having an updated version of this package would help with some other projects I'm working on. Thanks!

File: recipes/parasail-python/meta.yaml [hunk 3]
```diff
@@ -21,6 +24,7 @@ requirements:
     - pip
     - setuptools
     - {{ compiler('c') }}
```
Comment (bgruening) [inline recipes/parasail-python/meta.yaml]: In build should be only the compilers autoconf, automake, libtool and m4 I think.

Comment (jvkersch) [reply recipes/parasail-python/meta.yaml]: @bgruening That makes sense, thanks for pointing that out. I've updated the recipe to leave only the build tools under build, and moved the rest to host. I've also removed numpy as a build-time requirement.

**Comment by jvkersch** [comment]
@bgruening Ready for another round of reviews at your convenience. Thanks!

**Comment by bgruening** [comment]
Merci @jvkersch!

**Pull request merged**